src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -1191,67 +1216,139 @@ private boolean isSuspended()
+             * srcSeqNumLimit would not be updated (remain as -1) if the
+             * frame were either:
+             * a) an old frame of an adaptive stream OR
+             * b) a frame from an adaptive stream on a layer we're not using
+             * @return true if we're accepting this frame
              */
-            private int srcSeqNumLimit = -1;
-
-            /**
-             * The start source sequence number of this frame.
-             */
-            private int srcSeqNumStart = -1;
+            private boolean isAccepting()
+            {
+                return frameDesc.lastSequenceNumberKnown();
+            }
 
             /**
              * Defines an RTP packet filter that controls which packets to be
              * written into some arbitrary target/receiver that owns this
              * {@link SeenFrame}.
              *
-             * @param source the {@link FrameDesc} that the RTP packet belongs
-             * to.
              * @param pkt the packet for which to decide to accept
              * @return <tt>true</tt> to allow the specified packet/<tt>buffer</tt>
              * to be written into the arbitrary target/receiver that owns this
              * {@link SeenFrame} ; otherwise, <tt>false</tt>
              */
-            boolean accept(FrameDesc source, RawPacket pkt)
+            boolean accept(RawPacket pkt)
             {
-                if (this == mostRecentSentFrame /* the max frame can expand */
-                    || availableQualityIndices == null
-                    || availableQualityIndices.length < 2)
-                {
-                    int end = source.getEnd();
-                    srcSeqNumLimit
-                        = end != -1 ? source.getEnd() : source.getMaxSeen() ;
-                }
-
-                if (srcSeqNumLimit == -1)
+                if (!isAccepting())
                 {
                     return false;
                 }
-
-                int start = source.getStart();
-                srcSeqNumStart
-                    = start != -1 ? source.getStart() : source.getMinSeen();
-
                 int seqNum = pkt.getSequenceNumber();
 
-                boolean accept
-                    = RTPUtils.sequenceNumberDiff(seqNum, srcSeqNumLimit) <= 0;
+                // We'll accept the packet if it's no newer than our srcSeqNumLimit
+                // FIXME(brian): in what scenario will this test fail?  mostRecentSentFrame
+                // will always, at least, be equal to the frame to which this
+                // sequence number belongs, and we update srcSeqNumLimit based
+                // on the highest seen sequence number of that frame.  if seqNum
+                // was from a new frame
+                boolean accept =
+                    RTPUtils.isOlderSequenceNumberThan(
+                        seqNum,
+                        RTPUtils.applySequenceNumberDelta(frameDesc.getMaxSeen(), 1));
```

Comment:
i understand that, but `seqNum` comes from this frame, so `maxSeen` will have been updated based on this packet, meaning it can't possibly be newer than `maxSeen`, no?